pkg/getEc2Info.go

```diff
@@ -9,50 +9,96 @@ import (
 	"github.com/aws/aws-sdk-go-v2/service/ec2"
 )
 
-
 type EC2Instance struct {
+	InstanceName      string
+	InstanceID        string
+	InstanceType      string
+	InstancePrivateIP string
+}
 
-	InstanceName 		string
-	InstanceID	 		string
-	InstanceType 		string
-	InstancePrivateIP 	string
-
+func Ec2Client() (ec2.Client, context.Context) {
+	ctx := context.TODO()
+	cfg, err := config.LoadDefaultConfig(ctx)
+	if err != nil {
+		log.Fatal("Config Error Occured", err)
+	}
+	Ec2client := ec2.NewFromConfig(cfg)
+	return *Ec2client, ctx
 }
 
+func GetEC2Instance() []EC2Instance {
 
+	var ec2Instance []EC2Instance
+	client, ctx := Ec2Client()
+	input := &ec2.DescribeInstancesInput{}
+	info, err := client.DescribeInstances(ctx, input)
+	if err != nil {
+		log.Fatal("Error Occured while retrieving Information", err)
+	}
 
-func GetEC2Instance()([]EC2Instance){
-	var ec2Instance [] EC2Instance
+	instanceName := "-"
+	for _, r := range info.Reservations {
+		for _, i := range r.Instances {
+			for _, j := range i.Tags {
+				if *j.Key == "Name" {
+					instanceName = aws.ToString(j.Value)
+				}
+			}
+			ec2Instance = append(ec2Instance, EC2Instance{
+				InstanceName:      instanceName,
+				InstanceID:        aws.ToString(i.InstanceId),
+				InstanceType:      *aws.String(string(*&i.InstanceType)),
+				InstancePrivateIP: aws.ToString(*&i.PrivateIpAddress),
+			})
+		}
+	}
 
-	ctx := context.TODO()
-	cfg,err:=config.LoadDefaultConfig(ctx)
+	return ec2Instance
+}
 
-	if err!=nil{
-		log.Fatal("Config Error Occured",err)
-	}
+func GetEC2InstanceTag(tagName *string, tagValue []string) []EC2Instance {
 
-	client := ec2.NewFromConfig(cfg)
-	
 	input := &ec2.DescribeInstancesInput{}
-	info, err := client.DescribeInstances(ctx,input)
+
+	client, _ := Ec2Client()
+
+	info, err := client.DescribeInstances(context.TODO(), input)
 	if err != nil {
 		log.Fatal("Error Occured while retrieving Information", err)
 	}
-	
-	instanceName:="-"
+
+	instanceName := ""
+	var ec2Instance []EC2Instance
 	for _, r := range info.Reservations {
 		for _, i := range r.Instances {
-			for _,j:= range i.Tags{
-				if *j.Key == "Name"{
-					instanceName=aws.ToString(j.Value)
+			for _, j := range i.Tags {
```

Comment:
Can we somehow break this into multiple functions? 
these nested for loops are not readable.